components/infobox/wikis/starcraft/infobox_unit_custom.lua

```diff
@@ -0,0 +1,212 @@
+---
+-- @Liquipedia
+-- wiki=starcraft
+-- page=Module:Infobox/Unit/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local CostDisplay = require('Module:Infobox/Extension/CostDisplay')
+local Faction = require('Module:Faction')
+local Hotkeys = require('Module:Hotkey')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+
+local Injector = Lua.import('Module:Infobox/Widget/Injector', {requireDevIfEnabled = true})
+local Unit = Lua.import('Module:Infobox/Unit', {requireDevIfEnabled = true})
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Cell = Widgets.Cell
+local Title = Widgets.Title
+
+local CustomUnit = Class.new()
+
+local CustomInjector = Class.new(Injector)
+
+local ICON_HP = '[[File:Icon_Hitpoints.png|link=]]'
+local ICON_SHIELDS = '[[File:Icon_Shields.png|link=Plasma Shield]]'
+local ICON_ARMOR = '[[File:Icon_Armor.png|link=Armor]]'
+local UNKNOWN_RACE = 'u'
+
+local _args
+
+function CustomUnit.run(frame)
+	local unit = Unit(frame)
+	_args = unit.args
+
+	unit.nameDisplay = CustomUnit.nameDisplay
+	unit.setLpdbData = CustomUnit.setLpdbData
+	unit.getWikiCategories = CustomUnit.getWikiCategories
+	unit.createWidgetInjector = CustomUnit.createWidgetInjector
+
+	return unit:createInfobox()
+end
+
+function CustomInjector:addCustomCells(widgets)
+	return {
+		Title{name = 'Unit stats'},
+		Cell{name = 'Attributes', content = {_args.att}},
+		Cell{name = 'Defense', content = {CustomUnit:_defenseDisplay()}},
+		Cell{name = 'Damage', content = {_args.damage}},
+		Cell{name = 'Ground Damage', content = {_args.ground_attack}},
+		Cell{name = 'Air Damage', content = {_args.air_attack}},
+		Cell{name = '[[Distance#Range|Range]]', content = {_args.range}},
+		Cell{name = '[[Distance#Range|Ground Range]]', content = {_args.grange}},
+		Cell{name = '[[Distance#Range|Air Range]]', content = {_args.arange}},
+		Cell{name = '[[Distance#Range|Minimum Range]]', content = {_args.mrange}},
+		Cell{name = '[[Distance#Range|Minimum A. Range]]', content = {_args.marange}},
+		Cell{name = '[[Distance#Range|Minimum G. Range]]', content = {_args.mgrange}},
+		Cell{name = '[[Distance#Leash Range|Leash Range]]', content = {_args.lrange}},
+		Cell{name = '[[Game Speed#Cooldown|Cooldown]]', content = {_args.cooldown}},
+		Cell{name = '[[Game Speed#Cooldown|G. Cooldown]]', content = {_args.gcd}},
+		Cell{name = '[[Game Speed#Cooldown|A. Cooldown]]', content = {_args.acd}},
+		Cell{name = '[[Game Speed#Cooldown|Cooldown Bonus]]', content = {
+			_args.cd2 and _args.bonus2 and (_args.cd2 .. ' ' .. _args.bonus2)
+			or _args.cd2
+		}},
```

Comment:
The content part is repeated quite frequently, I'd make a local helper function to build the content string